dotnet/src/SemanticKernel.Abstractions/SemanticFunctions/PromptTemplateConfig.cs

```diff
@@ -152,6 +152,32 @@ public class InputConfig
     [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
     public InputConfig Input { get; set; } = new();
 
+    /// <summary>
+    /// If true, the prompt template will be considered trusted (default true).
+    /// This is just for the prompt itself, context variables such as the input
+    /// are handled separetely.
+    /// Untrusted prompts should only be allowed to run with not sensitive functions.
+    /// </summary>
+    [JsonPropertyName("is_prompt_trusted")]
+    [JsonPropertyOrder(7)]
+    public bool IsPromptTrusted { get; set; } = true;
+
+    /// <summary>
+    /// If true, the output of the function will always be considered to be untrusted
+    /// regardless of the input (default false).
+    /// </summary>
+    [JsonPropertyName("force_output_to_be_untrusted")]
+    [JsonPropertyOrder(8)]
+    public bool ForceOutputToBeUntrusted { get; set; } = false;
+
+    /// <summary>
+    /// Whether the function is set to be sensitive (default false).
+    /// Sensitive functions should not be allowed to run with untrusted input.
+    /// </summary>
+    [JsonPropertyName("is_sensitive")]
```

Comment:
I don't know if we can apply this flag to a prompt. Prompts can be sent to multiple backends, e.g. OpenAI, Azure OpenAI, local models, etc. and these have different trust characteristics

Reply:
Absolutely, the `IsSensitive` flag might not make sense directly linked to the prompt. I've just updated the code and moved this flag to the `SemanticFunctionConfig`. For native functions, it is available as part of the `SKFunctionAttribute` attribute.

Those can be changed in runtime in the current implementation if desired by the user, meaning this flag can be turned on and off by doing `functionInstance.IsSensitive = true|false`.